src/tests/baseservices/exceptions/simple/ParallelCrash.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+// Runtime stability in the presence of concurrent fatal errors
+
+public class ParallelCrash
+{
+    private const int ThreadCount = 10;
+
+    private volatile static int _runningThreads;
+
+    public static int Main()
+    {
+        Thread[] threadMap = new Thread[ThreadCount];
+        for (int threadIndex = 0; threadIndex < ThreadCount; threadIndex++)
+        {
+            Thread thread = new Thread(new ThreadStart(CrashInParallel));
+            threadMap[threadIndex] = thread;
+            thread.Start();
+        }
+        for (;;)
+        {
+            Thread.Sleep(1000);
+        }
+        return 0;
+    }
+
+    private static void CrashInParallel()
+    {
+        int threadIndex = ++_runningThreads;
```

Comment:
`Interlocked.Increment` should be used to avoid race condition.

Reply:
Also I would move the increment below the `string.Format` call.

Reply:
I have changed the increment operator to Interlocked.Increment in 4th commit but I left it above the string.Format call because I need the value returned from the method as the thread ID to print.